Networking/Networking/Network/NetworkError.swift

```diff
@@ -53,3 +54,38 @@ extension NetworkError {
         static let timeout  = 408
     }
 }
+
+extension NetworkError: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .notFound:
+            return NSLocalizedString(
+                "NetworkError.notFound",
+                value: "Sorry, we couldn't find what you were looking for. Please try again.",
+                comment: "Error message when we receive not found error")
+        case .timeout:
+            return NSLocalizedString(
+                "NetworkError.timeout",
+                value: "Sorry, the request took too long to process. Please try again later.",
+                comment: "Error message when a request times out.")
+        case let .unacceptableStatusCode(statusCode, responseData):
+            let response: String? = responseData.map { String(data: $0, encoding: .utf8) } ?? nil
+            let format = NSLocalizedString(
+                "NetworkError.unacceptableStatusCode",
+                value: "Sorry, there was an issue with the server. Please try again later. (Error code: %1$d). Response: %2$@",
+                comment: "Error message when the a network call fails with unacceptable status code." +
+                "%1$d is the status code like 500. %2$@ is the response string.")
+            return String.localizedStringWithFormat(format, statusCode, response ?? "")
+        case .invalidURL:
+            return NSLocalizedString(
+                "NetworkError.invalidURL",
+                value: "Sorry, the URL is not valid. Please try again.",
+                comment: "Error message when the URL is invalid.")
+        case .invalidCookieNonce:
+            return NSLocalizedString(
+                "NetworkError.invalidCookieNonce",
+                value: "Sorry, your session has expired. Please log in again.",
+                comment: "Error message when session cookie has expired.")
```

Comment:
nice user-facing strings 👍  I also wish we could have 2 separate strings for internal logging (non-localized) and user-facing. But I think we currently show the error's localized string in some UI.

Reply:
Good point. Now that we are including the response along with this string we will also be showing this in the UI. 🤔

Reply:
Hmm yea that's a great point 🤔 As a future consideration, maybe we can track the error description in a customized way like an extension for `NetworkError.loggingDescription`/`NetworkError.debugDescription` for Tracks events similar to https://github.com/woocommerce/woocommerce-ios/blob/3ce80118289ad2e106596ee6280d4b98c874c80c/WooCommerce/Classes/Analytics/WooAnalytics.swift#L157

However, in support tickets, lots of the errors are in `DDLog` that use `error.localizedDescription`. Maybe we'd need to make another protocol for errors for logging purposes with `localizedDescription` as the default.